cli/src/cli-rpc-ops.c

```diff
@@ -1970,6 +1970,60 @@ gf_cli_ganesha_cbk(struct rpc_req *req, struct iovec *iov, int count,
     return ret;
 }
 
+static char *
+added_user_xlator(void *myframe)
+{
+    call_frame_t *frame = NULL;
+    cli_local_t *local = NULL;
+    char **words = NULL;
+    char *key = NULL;
+    char *value = NULL;
+    char *added_xlator = NULL;
+    int i = 0;
+
+    /*
+     * HACK:
+     *  The server xlator list is also defined here, but it should
+     *  have a common definition with glusterd. However, we are hesitant
+     *  to include the internal header of glusterd.  If want to use the
+     *  shared definition, we need to make a big modification.
+     */
+    static char *server_xlators[] = {
+        "posix", "arbiter", "trash",     "changelog",  "bitrot-stub", "acl",
+        "locks", "leases",  "upcall",    "io-threads", "selinux",     "marker",
+        "index", "quota",   "namespace", "sdfs"};
+
+    frame = myframe;
+    local = frame->local;
+    words = (char **)local->words;
+
+    while (*words != NULL) {
+        if (fnmatch("user.xlator.*", *words, 0) == 0 &&
+            fnmatch("user.xlator.*.*" /* user xlator option key*/, *words, 0) ==
+                FNM_NOMATCH) {
+            key = *words;
+            words++;
+            value = *words;
+
+            if (value == NULL)
+                break;
+
+            int size = sizeof(server_xlators) / sizeof(server_xlators[0]);
+
+            for (i = 0; i < size; i++) {
+                if (strcmp(value, server_xlators[i]) == 0) {
+                    added_xlator = gf_strdup(key);
+                    break;
+                }
+            }
+        }
+
+        words++;
+    }
+
+    return added_xlator;
```

Comment:
We don't distinguish between an error and no `user.xlator.*` option found. I think we should handle the error case so that the operation is reported as a failure instead of a success.

Reply:
I added the error handling and pushed it.
Please review.